cms/djangoapps/contentstore/views/course.py

```diff
@@ -286,7 +343,54 @@ def course_handler(request, course_key_string=None):
                     course_module = get_course_and_check_access(course_key, request.user, depth=None)
                     return JsonResponse(_course_outline_json(request, course_module))
             elif request.method == 'POST':  # not sure if this is only post. If one will have ids, it goes after access
-                return _create_or_rerun_course(request)
+                if request.POST.get('export_course') != 'true':
+                    return _create_or_rerun_course(request)
+
+                # Save a snapshot of the database for later processing.
+                course_key = CourseKey.from_string(course_key_string)
+                course_module = get_course_and_check_access(course_key, request.user, depth=1)
+                course_outline = _course_outline_json(request, course_module)
+                key_to_extract = ['id', 'edited_on']
+                snap_json = {key: course_outline[key] for key in course_outline if key in key_to_extract}
+                snap_json['chapter'] = [ 
+                    {
+                        'id':item.get('id'),
+                        'edited_on':item.get('edited_on'),
```

Comment:
please break down into modular code and move to edly app

Reply:
done